triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceTest.scala

```diff
@@ -616,6 +642,157 @@ trait AbstractTriggerServiceTest extends AbstractTriggerServiceTestHelper {
+      _ <- assertTriggerStatus(catsTrigger, _ should contain("stopped: runtime failure"))
+      _ <- assertTriggerRunnerStatus(
+        catsTrigger,
+        _ should contain(
+          "[ERROR] Due to an excessive number of active contracts, stopping the trigger"
+        ),
+      )
+    } yield succeed
+  }
+
+  it should "stop the trigger if the ACS overflows at runtime" inClaims withTriggerService(
+    List(dar),
+    triggerRunnerConfig = Some(
+      DefaultTriggerRunnerConfig
+        .copy(
+          // As the trigger creates 100 Cat contracts, we should eventually overflow using 1
+          maximumActiveContracts = 1
+        )
+    ),
+  ) { uri: Uri =>
+    for {
+      client <- sandboxClient(
+        catsAppId,
+        actAs = List(ApiTypes.Party(alice.unwrap)),
+      )
+      _ <- getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+        .flatMap(events =>
+          Future.sequence(events.map { event =>
+            submitCmd(client, alice.unwrap, killCat(event.contractId))
+          })
+        )
+      // Wait until there are no Cat contracts
+      _ <- RetryStrategy.constant(20, 1.seconds) { (_, _) =>
+        getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+          .map(_ shouldBe Vector())
+      }
+      resp <- startTrigger(uri, s"$testPkgId:Cats:breedingTrigger", alice, Some(catsAppId))
+      catsTrigger <- parseTriggerId(resp)
+      _ <- assertTriggerIds(uri, alice, Vector(catsTrigger))
+      _ <- assertTriggerStatus(catsTrigger, _ should contain("stopped: runtime failure"))
+      _ <- assertTriggerRunnerStatus(
+        catsTrigger,
+        _ should contain(
+          "[ERROR] Due to an excessive number of active contracts, stopping the trigger"
+        ),
+      )
+    } yield succeed
+  }
+
+  it should "stop the trigger if the in-flight commands overflow at runtime" inClaims withTriggerService(
+    List(dar),
+    triggerRunnerConfig = Some(
+      DefaultTriggerRunnerConfig.copy(
+        // Rate limit ledger submissions to 1 per millisecond
+        maxSubmissionDuration = 100.millis,
+        // As our submission rate is faster than the ledger can manage and we are submitting 100 Cat create commands,
+        // in-flights should overflow using 10
+        inFlightCommandOverflowCount = 10,
+      )
+    ),
+  ) { uri: Uri =>
+    for {
+      client <- sandboxClient(
+        catsAppId,
+        actAs = List(ApiTypes.Party(alice.unwrap)),
+      )
+      _ <- getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+        .flatMap(events =>
+          Future.sequence(events.map { event =>
+            submitCmd(client, alice.unwrap, killCat(event.contractId))
+          })
+        )
+      // Wait until there are no Cat contracts
+      _ <- RetryStrategy.constant(20, 1.seconds) { (_, _) =>
+        getActiveContracts(client, alice, Identifier(testPkgId, "Cats", "Cat"))
+          .map(_ shouldBe Vector())
+      }
```

Comment:
Pollution from prior testing requires us to cleanup prior to allowing this test to proceed